data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDB.java

```diff
@@ -383,4 +389,31 @@ public void deleteEnvironmentRevision(@NonNull final EnvironmentRevision environ
           e);
     }
   }
+
+  @Override
+  public List<Cluster> listClusters(String accountId, String clusterNamePrefix) {
+    if (clusterNamePrefix != null) {
+      Validate.notNull(accountId, "accountId must be specified if clusterNamePrefix is specified");
+    }
+
+    // TODO: rewrite this to use the index instead of scanning the table
+    PaginatedScanList<EnvironmentDDBRecord> records =
+        dynamoDBMapper.scan(EnvironmentDDBRecord.class, new DynamoDBScanExpression());
```

Comment:
We're doing a full table scan to get all clusters. Do we know how slow this is in the extreme case?

Reply:
See the review description; this is a short-term hack, working on fixing the data model/index now.

Reply:
Makes sense....I should have read the description first :-D Waiting for your next PR